pandas/tseries/period.py

```diff
@@ -281,6 +281,16 @@ def __contains__(self, key):
             return False
         return key.ordinal in self._engine
 
+    def __array_wrap__(self, result, context=None):
```

Comment:
Reconsidered this, and defining ``_add_ndarray`` for ``PeriodIndex`` may be an alternative. I think options are:

1. Always raise ``TypeError`` in ``__array_wraps__``. Support ``ndarray`` ops in another methods (``PeriodIndex`` must be lhs).

2. Raise ``TypeError`` if ``__array_wraps__`` gets non-integers. Some ``ufunc`` which return ``int`` outputs meaningless results (like ``square``)



 Maybe not intentional, ``PeriodIndex.shift`` works for ``ndarray``.